docs/configuration/config-file-reference.md

```diff
@@ -2587,14 +2587,6 @@ The `limits_config` configures default and per-tenant limits imposed by Cortex s
 # CLI flag: -validation.max-metadata-length
 [max_metadata_length: <int> | default = 1024]
 
-# Reject old samples.
-# CLI flag: -validation.reject-old-samples
-[reject_old_samples: <boolean> | default = false]
```

Comment:
Per https://cortexmetrics.io/docs/configuration/v1guarantees/#flags-config-and-minor-version-upgrades we might need to keep the `reject_old-*` config for 2 minor releases.

Reply:
I see. I will add them back.

Reply:
I am wondering about the behaviour of the co-existence of `reject_old_samples`  and `out_of_order_time_window`. 

By default `reject_old_samples=false` but `out_of_order_time_window=0`, one config says accept old (out of order) samples, the other says disable out of order samples. What should Cortex do? 

Generally speaking, what happens if the two config seemingly conflicts with each other? May be worth to document the behaviour in `v1-guarantees.md`.

Reply:
For the default values, if the sample is too old it will be rejected by the TSDB anyway if we don't enable OOO. This is the same even if we don't have OOO support so we don't change behavior here. 
If `reject_old_samples=false` means `accept old (out of order) samples` then I believe our document was wrong. It just means we allow those samples on the distributor side

Reply:
Actually I feel instead of adding this to `v1-guarantees.md`, it is better to document this to the Out of order samples operational doc you mentioned? 
Wdyt? `v1-guarantees.md` seems just list flags we have, but nothing really detailed about the settings and usage.

Reply:
Im not sure if we should deprecate this flag....

Could i configure i can accept our of order for 5 min BUT if not out of order i can accept samples that is 2 hours old?

Reply:
> Could i configure i can accept our of order for 5 min BUT if not out of order i can accept samples that is 2 hours old?

Isn't it doable with OOO only? Non out of order samples are always accepted unless it is outside head time range.

But I agree they are still slightly different things since we can drop samples early on distributors rather than waiting till ingesters.

Reply:
Discussed with @alanprot, we decided to not deprecate the two flags as they are different from the OOO settings.